### ISSUE TITLE
mosaic augmentation

### ISSUE THREAD

**Issue by joe660** [ultralytics/yolov5#1423]
## ❔Question
Hello, use your own data set to train on yolov5. For some reasons, you need to turn off mosaic augmentation to get some important information. May I ask how much the removal of mosaic augmentation affects the performance of the model.

## Additional context
Hello, use your own data set to train on yolov5. For some reasons, you need to turn off mosaic augmentation to get some important information. May I ask how much the removal of mosaic augmentation affects the performance of the model.

**Comment by AzharSindhi** [comment]
I think this is a good question, anyone has some insights?

**Comment by glenn-jocher** [comment]
@AzharSindhi in general training will perform worse without mosaic augmentation. Recommend you train with all default settings, including default mosaic setting.

**Comment by alevangel** [comment]
I would like to remove the mosaic augmentation, also the scaling augmentation. How can I do that?

**Comment by glenn-jocher** [comment]
@alevangel you can modify hyperparameters in your hyperparameters file:
https://github.com/ultralytics/yolov5/blob/e96c74b5a1c4a27934c5d8ad52cde778af248ed8/data/hyps/hyp.scratch.yaml#L1-L34

**Comment by Tautvydas-byte** [comment]
Hello, How to check how much new images were created and gave to the model after augmentation ? For example I give 117 images and using default mosaic augmentation with default hyper-params.

**Comment by glenn-jocher** [comment]
@Tautvydas-byte 👋 Hello! Thanks for asking about **image augmentation**. YOLOv5 🚀 applies online imagespace and colorspace augmentations in the trainloader (but not the val_loader) to present a new and unique augmented [Mosaic](https://github.com/ultralytics/yolov5/blob/90b7895d652c3bd3d361b2d6e9aee900fd67f5f7/utils/datasets.py#L678-L732) (original image + 3 random images) each time an image is loaded for training. **Images are never presented twice in the same way**.

![YOLOv5 augmentation](https://user-images.githubusercontent.com/26833433/120995721-f3cfed00-c785-11eb-8ee2-b6ef2fa205e8.jpg)

### Augmentation Hyperparameters

The hyperparameters used to define these augmentations are in your hyperparameter file (default `data/hyp.scratch.yaml`) defined when training:
```
python train.py --hyp hyp.scratch-low.yaml
```
https://github.com/ultralytics/yolov5/blob/b94b59e199047aa8bf2cdd4401ae9f5f42b929e6/data/hyps/hyp.scratch-low.yaml#L6-L34

### Augmentation Previews

You can view the effect of your augmentation policy in your train_batch*.jpg images once training starts. These images will be in your train logging directory, typically `yolov5/runs/train/exp`:

`train_batch0.jpg` shows train batch 0 mosaics and labels:  
<img src="https://user-images.githubusercontent.com/26833433/131255960-b536647f-7c61-4f60-bbc5-cb2544d71b2a.jpg" width="800">


### YOLOv5 Albumentations Integration

[YOLOv5](https://github.com/ultralytics/yolov5) 🚀 is now fully integrated with [Albumentations](https://github.com/albumentations-team/albumentations), a popular open-source image augmentation package. Now you can train the world's best Vision AI models even better with custom Albumentations 😃! 

PR https://github.com/ultralytics/yolov5/pull/3882 implements this integration, which will automatically apply Albumentations transforms during YOLOv5 training if `albumentations>=1.0.3` is installed in your environment. See https://github.com/ultralytics/yolov5/pull/3882 for full details.

Example `train_batch0.jpg` on COCO128 dataset with Blur, MedianBlur and ToGray. See the YOLOv5 [Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb) to reproduce: <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>

<img src="https://user-images.githubusercontent.com/26833433/124400879-ff331b80-dd25-11eb-9b67-fe85ac4ca104.jpg" width="800">

Good luck 🍀 and let us know if you have any other questions!

**Comment by Tautvydas-byte** [comment]
Thank you, for your fast answer. Speaking about augmentation what we see in the mosaic. Only these what we can see for example in the train_batch0.jpg is giving to model for every epoch or mosaic is changing in every epoch?

**Comment by glenn-jocher** [comment]
@Tautvydas-byte there's no such thing as two identical mosaics during training.
